tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+    verison of tsfresh as defined by :mod:`tsfresh_feature_names.TSFRESH_FEATURES`
+
+    .. warning:: the Python 2 and 3 calculate different results in terms of
+        float precision therefore baseline transposed features csv files are
+        required for both py2 and py3.
+
+    Running the test
+    ================
+
+    .. code-block:: bash
+
+        cd "<YOUR_TSFRESH_DIR>"
+        python -m pytest tests/baseline/tsfresh_features_test.py
+
+
+    Test the test fails
+    ===================
+
+    To test that the test fails as desired and as does what it is supposed to do
+    there are 2 methods to achieve this:
+
+    - Modify the first value in your local tsfresh/examples/data/test_tsfresh_baseline_dataset/data.json
+      and run the test, then delete the modified local data.json file to be pulled down again.
+    - Modify a feature name or value in your local tests/baseline/tsfresh-<TSFRESH_BASELINE_VERSION>.py<PYTHON_VERSION>.data.json.features.transposed.csv file,
+      run the test and either pull it again or revert the change
+
+    """
+    def setUp(self):
+        self.baseline_ts_json = download_json_dataset()
+
+    def tearDown(self):
+        # Remove the directory after the test
+        try:
+            shutil.rmtree(test_path)
+        except:
+            pass
+
+    def test_tsfresh_baseline(self):
+        """
+        This test should generates a set of resources from a sample 60 data
+        point timeseries, extracts the features and saves a transposed csv of
+        feature_name,value that is used to compare against to baseline
+        transposed csv
+        """
+        baseline_ts_json = self.baseline_ts_json
+
+        test_path_created = False
+        if os.path.exists(test_path):
+            test_path_created = True
+        self.assertEqual(test_path_created, True)
+
+        json_data_exists = False
+        if os.path.isfile(baseline_ts_json):
+            json_data_exists = True
+
+        self.assertEqual(json_data_exists, True)
+
+        timeseries_json = None
+        if os.path.isfile(baseline_ts_json):
+            with open(baseline_ts_json, 'r') as f:
+                timeseries_json = json.loads(f.read())
+
+        if python_version == 2:
+            timeseries_str = str(timeseries_json).replace('{u\'results\': ', '').replace('}', '')
+        if python_version == 3:
+            timeseries_str = str(timeseries_json).replace('{\'results\': ', '').replace('}', '')
+
+        full_timeseries = eval(timeseries_str)
+        timeseries = full_timeseries[:60]
+        self.assertEqual(int(timeseries[0][0]), 1369677886)
+        self.assertEqual(len(timeseries), 60)
+
+        for ts, value in timeseries:
+            metric = 'tsfresh_features_test'
+            timestamp = int(ts)
+            value = str(float(value))
+            utc_ts_line = '%s,%s,%s\n' % (metric, str(timestamp), value)
+            with open(tmp_csv, 'a') as fh:
+                fh.write(utc_ts_line)
```

Comment:
The data only has to be fiddled around with to get it into an acceptable tsfresh data format :)

Saving it as CSV does not follow the same methodology that tsfresh uses in other tests related to public data sources.  And seeing as it is a port over from my tests patterns from Skyline, that would mean I have to change the data source here, etc.

Personally I think a little string manipulation is not so offence.

Going from methodology from other tsfresh tests, it appears if they all pull data from sources, only in the testing context, original I did just have the data.json in the directory but I figured perhaps you did not want data sets bulking up the repo.  Further to that, should any more energetic timeseries data be added to this test, then this external public source method only if testing is quite sensible way to handle the data.  That said I think there should probably always be at least 2 available copies/urls for the data sources.  Perhaps a tsfresh-data repo could be useful too ensure that their is always a failback data source if x external source is unavailable for some reason.